### PR TITLE
Ui/transform roles

### PR DESCRIPTION
**Pull request by chelshaw**
This PR sets up the items within the transform secrets engine, and lists each of them as read-only items. 

This is in draft mode because I'd like to get the create roles flow in working order before we merge this

transform-alphabets
transform-roles-empty
transform-templates
transform-roles

### CODE REVIEW

File: ui/app/routes/vault/cluster/secrets/backend/list.js [hunk 1]
```diff
@@ -22,6 +22,25 @@ export default Route.extend({
     },
   },
 
+  modelTypeForTransform(tab) {
```
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/secrets/backend/list.js:25]: nice!

File: ui/app/routes/vault/cluster/secrets/backend/list.js [hunk 2]
```diff
@@ -83,6 +102,7 @@ export default Route.extend({
           pageFilter: params.pageFilter,
         })
         .then(model => {
+          console.log({ model });
```
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/secrets/backend/list.js:105]: just a heads up about the console log.  I know this is still a draft, but wanted to point it out.

File: ui/app/templates/partials/secret-list/transform-transformation-item.hbs [hunk 3]
```diff
@@ -1,60 +1,75 @@
...
+        <Icon
+          @glyph='file-outline'
+          @class="has-text-grey-light"/>
+        {{if (eq item.id ' ') '(self)' (or item.keyWithoutParent item.id)}}
+      </div>
+    </div>
   </div>
-</div>
-{{/linked-block}}
+{{/if}}
```
Comment (Monkeychip) [inline ui/app/templates/partials/secret-list/transform-transformation-item.hbs:75]: missing space at end of file.

**Comment by chelshaw** [comment]
Did a big rebase to put most of this work on top of the updated main branch instead of a sidebranch. Will open a new PR when that's ready for review

**Pull request closed without merging**